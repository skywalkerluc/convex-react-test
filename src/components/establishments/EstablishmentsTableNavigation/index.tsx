import React from 'react';
// import styles from './styles.module.css';

interface NavigationProps {
  currentPage: number;
  totalPages: number;
  onPrevious: () => void;
  onNext: () => void;
  loading: boolean;
}

const EstablishmentsTableNavigation: React.FC<NavigationProps> = ({
  currentPage,
  totalPages,
  onPrevious,
  onNext,
  loading
}) => {
  const isFirstPage = currentPage === 1;
  const isLastPage = currentPage === totalPages;

  return (
    <div data-testid="pagination-controls">
      <button
        type="button"
        onClick={onPrevious}
        disabled={loading || isFirstPage}
        aria-disabled={loading || isFirstPage}
        aria-label="Previous page"
      >
        Previous
      </button>

      <span>
        Page {currentPage} of {totalPages}
      </span>

      <button
        type="button"
        onClick={onNext}
        disabled={loading || isLastPage}
        aria-disabled={loading || isLastPage}
        aria-label="Next page"
      >
        Next
      </button>
    </div>
  );
};

export default React.memo(EstablishmentsTableNavigation);