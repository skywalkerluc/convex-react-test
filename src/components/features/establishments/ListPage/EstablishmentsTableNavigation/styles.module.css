.navigationContainer {
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 2rem;
    margin-top: 2rem;
    padding: 1rem;
  }
  
  .button {
    padding: 0.5rem 1.5rem;
    background-color: #235d45;
    color: rgba(255, 255, 255, 0.9);
    border: none;
    border-radius: 4px;
    cursor: pointer;
    transition: background-color 0.2s;
  }
  
  .button:disabled {
    background-color: #6c757d;
    cursor: not-allowed;
  }
  
  .button:hover:not(:disabled) {
    background-color: #11291f;
  }
  
  .pageInfo {
    font-weight: 500;
    color: #f0f0f0;
  }