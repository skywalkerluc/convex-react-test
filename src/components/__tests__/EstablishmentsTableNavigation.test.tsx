import { render } from '@testing-library/react';
import EstablishmentsTableNavigation from '../establishments/EstablishmentsTableNavigation';

describe('EstablishmentsTableNavigation', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <EstablishmentsTableNavigation
        currentPage={2}
        totalPages={5}
        onPrevious={() => {}}
        onNext={() => {}}
        loading={false}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});