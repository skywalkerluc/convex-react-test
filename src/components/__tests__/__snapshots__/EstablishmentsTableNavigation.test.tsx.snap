// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EstablishmentsTableNavigation matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="navigationContainer"
    data-testid="pagination-controls"
  >
    <button
      aria-disabled="false"
      aria-label="Previous page"
      class="button"
      type="button"
    >
      Previous
    </button>
    <span
      class="pageInfo"
    >
      Page 2 of 5
    </span>
    <button
      aria-disabled="false"
      aria-label="Next page"
      class="button"
      type="button"
    >
      Next
    </button>
  </div>
</DocumentFragment>
`;
